### PR TITLE
Fix editoast layer cache invalidation

### LINKED ISSUE #5931: Layer invalidation does not work properly
### What happened?

A series of specific actions can lead to a wrong state of cache in redis.

### What did you expect to happen?

Redis cache should always be up to date.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to editor
2. Selec a track section with a buffer stop on it
3. Delete the track section
4. The buffer stop is still visible at some level of zoom

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

3df6fbc

### PR DESCRIPTION
**Pull request by flomonster**
close #5931

### CODE REVIEW

File: editoast/src/map/mod.rs [hunk 1]
```diff
@@ -61,64 +37,61 @@ async fn invalidate_cache_tiles<C: ConnectionLike>(
...
-/// If the zone is invalide nothing is done
+/// Invalidates layer cache for a specific infra and view if provided
 ///
 /// # Arguments
 ///
 /// * `redis_pool` - Pool to use to connect to the redis
-/// * `layers` - Layers to invalidate
-/// * `infra_id` - Infra to on which layers must be invalidated
-/// * `zone` - Zone to invalidate
+/// * `infra_id` - Infra to on which the layer must be invalidated
```
Comment (Erashin) [inline editoast/src/map/mod.rs]:
```suggestion
/// * `infra_id` - Infra on which the layer must be invalidated
```

**Review by Erashin — APPROVED** [review]
LGTM

File: editoast/src/map/mod.rs [hunk 2]
```diff
@@ -61,64 +37,61 @@ async fn invalidate_cache_tiles<C: ConnectionLike>(
...
-    map_config: &MapLayersConfig,
-) -> Result<()> {
-    if zone.is_valid() {
-        for layer in layers {
-            invalidate_layer_zone(redis, infra_id, layer, zone, map_config).await?;
-        }
-    }
-    Ok(())
+    layer_name: &str,
+    view_name: Option<&str>,
```
Comment (woshilapin) [inline editoast/src/map/mod.rs]: Having `Option` as parameters is a weird pattern: either you call the function with something or you don't. In this case, it seems that there is a case for 2 distinct functions: `get_prefix` and `invalidate_full_layer_cache(layer_name: &str, prefix: &str)`.

Some reading about that pattern https://matklad.github.io/2023/11/15/push-ifs-up-and-fors-down.html.

**Review by woshilapin — APPROVED** [review]
A comment, opinionated on some pattern. I'm good to merge as-is, without addressing it if you think it useless. Apart from that, not sure I understand everything 😅 but looks good!

File: editoast/src/map/mod.rs (hunk 2 above)
Comment (flomonster) [reply editoast/src/map/mod.rs]: I split the function in two :wink:

**Review by woshilapin — APPROVED** [review]

File: editoast/src/map/mod.rs [hunk 3]
```diff
@@ -61,64 +37,79 @@ async fn invalidate_cache_tiles<C: ConnectionLike>(
...
 ///
 /// # Arguments
 ///
 /// * `redis_pool` - Pool to use to connect to the redis
 /// * `infra_id` - Infra to on which the layer must be invalidated
 /// * `layer_name` - Layer on which invalidation must be done
-/// * `zone` - Zone to invalidate
-async fn invalidate_layer_zone<C: ConnectionLike>(
+/// * `view_name` - Layer on which invalidation must be done
+/// * `bbxo` - Bounding Box to invalidate (full view if None)
```
Comment (leovalais) [inline editoast/src/map/mod.rs]:
```suggestion
/// * `bbox` - Bounding Box to invalidate (full view if None)
```

**Review by leovalais — APPROVED** [review]
LGTM, tested

**Comment by Tguisnet** [comment]
A bit late but LGTM

**Pull request merged**